vscode-extension/src/aiConfigEditor.ts

```diff
@@ -251,6 +251,62 @@ export class AIConfigEditorProvider implements vscode.CustomTextEditorProvider {
         case "open_in_text_editor":
           vscode.commands.executeCommand("workbench.action.reopenTextEditor");
           return;
+
+        case "show_notification":
+          const notification = e.notification as {
+            // AIConfigEditorNotification
+            title: string;
+            message: string | null;
+            type?: "info" | "success" | "warning" | "error";
+            autoClose?: boolean | number;
+            onClose?: () => void;

```

Comment:
Do onOpen and onClose work in this context? These functions are supplied from the webview?

Reply:
Ya, the editor component supports specifying onOpen and onClose as part of the notification object (which is also handled by mantine `showNotification`). In vscode case, we kind of implicitly handle it -- onOpen is just before the notification is called and onClose happens once the notification is resolved (happens on explicit close or button press). `autoClose` isn't supported directly by the vs code notification api but I've seen some hacks to support it - just prefer not to implement it until it's needed since it's hard to test with the current package scenario